pkg/generate/template.go

```diff
@@ -0,0 +1,67 @@
+package generate
+
+import (
+	"errors"
+	"io"
+	"io/ioutil"
+	"regexp"
+	"strings"
+)
+
+var (
+	ErrCouldNotReadFile = errors.New("could not read file")
+)
+
+// Template contains an array of documents that can be
+// interacted with with its various functions.
+type template struct {
+	documents []string
+}
+
+func buildTemplate(file io.Reader) (template, error) {
+	// Separate the documents by the yaml separator and build a template with them
+	fileBytes, err := ioutil.ReadAll(file)
+	if err != nil {
+		return template{}, ErrCouldNotReadFile
+	}
+
+	docs := strings.Split(string(fileBytes), "---")
+
+	constructedTemplate := template{}
+	for _, doc := range docs {
+		trimmedDoc := strings.TrimSpace(doc)
+		if trimmedDoc != "" {
+			constructedTemplate.documents = append(constructedTemplate.documents, trimmedDoc)
+		}
+	}
+
+	return constructedTemplate, nil
+}
+
+// has determines if any of the documents for the template
+// contains the specified string.
+func (t template) has(find string) bool {
```

Comment:
nit: `find` reads weird for me here. Would something like `str` or `searchString` or something like that?